config/envoyconfig/tls.go

```diff
@@ -24,6 +25,17 @@ func (b *Builder) buildSubjectAltNameMatcher(
 		sni = overrideName
 	}
 
+	if net.ParseIP(sni) != nil {
```

Comment:
I vaguely recall something about the netip package being more efficient? https://pkg.go.dev/net/netip#ParseAddr

Reply:
(I think maybe that was only for memory if you actually use IPs in code, so probably makes no difference.

Reply:
Sure, it probably makes sense to use the newer package. I see the netip package also supports [scoped IPv6 addresses](https://en.wikipedia.org/wiki/IPv6_address#Scoped_literal_IPv6_addresses_.28with_zone_index.29). I won't pretend to fully understand this concept, but I figure if someone uses a scoped IPv6 address in their route's `to` URL, it probably makes sense to recognize that as an IP address.

If a zone _is_ present I believe we need to strip that from the SAN matcher (IPv6 SAN's cannot contain a zone, if I'm reading https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6 correctly). I've added another unit test case for this.

Of course I really hope no one is creating certificates with link-local IPv6 addresses in the first place.